### PR TITLE
feat(misconf): Add support for deprecating a check

### PR DESCRIPTION
**Pull request by simar7**
## Description
Add support for deprecating a check.

## Related issues
- Implements: https://github.com/aquasecurity/trivy/issues/6626

## Related PRs
- https://github.com/aquasecurity/trivy/pull/6671

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by simar7** [comment]
@nikpivkin I'll implement the support for the `--ignore-deprecated-checks` in another PR to keep the PRs small. Here's the issue for that https://github.com/aquasecurity/trivy/issues/6665

File: pkg/flag/rego_flags.go [hunk 1]
```diff
@@ -7,6 +7,11 @@ package flag
 //	  config-policy: "custom-policy/policy"
 //	  policy-namespaces: "user"
 var (
+	IncludeDeprecatedChecksFlag = Flag[bool]{
```
Comment (nikpivkin) [inline pkg/flag/rego_flags.go:10]: The [config-file.md](https://github.com/aquasecurity/trivy/blob/main/docs/docs/references/configuration/config-file.md) should be updated.

**Comment by nikpivkin** [comment]
Left a small comment, but lgtm.

File: pkg/flag/rego_flags.go (hunk 1 above)
Comment (simar7) [reply pkg/flag/rego_flags.go:10]: Thanks, updated.

**Review by nikpivkin — APPROVED** [review]

**Pull request merged**